### PR TITLE
Document tchannel service to be implemented by client libraries

### PR DESCRIPTION
**Pull request by prashantv**
I imagine introspection APIs would also live under this tchannel service.

r @blampe @Raynos @kriskowal @jcorbin

### CODE REVIEW

File: thrift/tchannel.thrift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+struct ClientLibrary {
```
Comment (blampe) [inline thrift/tchannel.thrift]: Why not in meta?

Comment (Raynos) [inline thrift/tchannel.thrift]: Here's an idea that might help with Meta vs TChannel

Meta is something that either:

 - The application developer can customize
 - Contains optional things.

The TChannel api is something that will be:

 - Implemented by tchannel without user customization
 - Does not contain optional features at all.

For example:

 - Meta health is customized by users so belongs in Meta
 - Meta thriftIdl only applies to thrift encoding so belongs in Meta
 - TChannel library version is not customized and is mandatory so belongs in tchannel introspection.

File: thrift/tchannel.thrift [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  1: required string language
+  2: required string version
+}
+
+service TChannel {
+  ClientLibrary clientLibrary()
```
Comment (Raynos) [inline thrift/tchannel.thrift]: Consider `VersionInfo versionInfo()`

**Comment by Raynos** [comment]
:+1: I dont care about naming that much...

File: thrift/tchannel.thrift (hunk 1 above)
Comment (prashantv) [inline thrift/tchannel.thrift]: Was thinking about that - if one Channel is registering on multiple Hyperbahn services, meta is registered on each of the service names (and the user may want custom health checks for each of those).

This seems more like global TChannel state that only needs to be registered once. However, it doesn't hurt to put this in meta, so happy to move it there if that's preferred.

E.g. it's the difference between exposing these endpoints (if it's part of Meta):
```
s1 Meta::health
s1 Meta::clientLibrary
s2 Meta::health
s2 Meta::clientLIbrary
tchannel Meta::health
tchannel Meta::clientLibrary
```
vs exposing these endpoints
```
s1 Meta::health
s2 Meta::health
tchannel Meta::clientLibrary
```

File: thrift/tchannel.thrift (hunk 2 above)
Comment (blampe) [inline thrift/tchannel.thrift]: +1

File: thrift/tchannel.thrift (hunk 1 above)
Comment (blampe) [inline thrift/tchannel.thrift]: Ok my only concern then is that we (humans) would basically have to address workers individually to discover a service's version info?

Comment (Raynos) [inline thrift/tchannel.thrift]: Endoints are cheap. Might as well put everything in meta and expose meta on all service names including tchannel.

**Comment by prashantv** [comment]
Renamed to `versionInfo` and moved to `Meta`.

**Pull request merged**